### PR TITLE
chore: Add Conventional commits Git message hook

### PR DESCRIPTION
**Pull request by mcdurdin**
Adds support for conventional commit style validation of all commit messages and makes configuration against all Keyman repos straightforward with the `configure-repo.sh` script.

### CODE REVIEW

**Comment by darcywong00** [comment]
Can we document the configure step in either the workflow [wiki](https://github.com/keymanapp/keyman/wiki/Pull-Request-and-Commit-workflow-notes) or each platform's README.md?

File: resources/git-hooks/commit-msg [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#
+# Reference: https://github.com/keymanapp/keyman/wiki/Pull-Request-and-Commit-workflow-notes
+#
+
+#
+# Configurable options for types, scopes and message length.
+#
+
+types=(fix feat chore change docs style refactor test)
+scopes=(android ci common developer ios linux mac web windows)
```
Comment (darcywong00) [inline resources/git-hooks/commit-msg]: A couple questions about scopes.

1. do we lump `lm-layer` with `web`?
2. what about `keyboardprocessor`, `oem`, and `resources`?

**Comment by jahorton** [comment]
Is the hook activated yet?  What should I look for on this PR to see the difference?

**Comment by mcdurdin** [comment]
@darcywong00 @jahorton see https://github.com/keymanapp/keyman/wiki/How-to-setup-conventional-commits-git-commit-msg-hook for details on activating the hook on your dev environment. Note that this will apply only to `master`, because the link will be missing if you check out other branches (once this PR merges).

If you wish, you can maintain the commit hook yourself by updating the git `core.hooksPath` setting and copying the `resources/git-hooks/commit-msg` script into the path you specify there. That can let you do other things.

File: resources/git-hooks/commit-msg (hunk 1 above)
Comment (mcdurdin) [reply resources/git-hooks/commit-msg]: * Yes, `lm-layer` goes with `web` (if you see the label in GitHub, it is now `web-lm-layer`).
* `keyboardprocessor` and `resources` go in `common`. 
* For `oem` use the platform that is being updated. If multiple platforms, you may want to go with `common`

I've added most of this to the wiki.

File: configure-repo.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# Setup git hooks in this or all repositories.
+#
+# Run this script once to configure your system to work with our Git workflow.
+#
+
+case $1 in
+  --global)
+    # Requires Git 2.9 or later.
+    coreHooksPath=`git config --global core.hooksPath`
+    if [[ -z "$coreHooksPath" ]]; then coreHooksPath="(not set)"; fi
```
Comment (darcywong00) [inline configure-repo.sh]: On my Linux box, I get the error
```
./configure-repo.sh: 13: ./configure-repo.sh: [[: not found
```

bash syntax?

Comment (darcywong00) [reply configure-repo.sh]: btw even with the error, the rest of the script ran

Comment (mcdurdin) [reply configure-repo.sh]: hmm. I dunno. Which version of bash do you run? Should be recent I'd have thunk.

Comment (mcdurdin) [reply configure-repo.sh]: https://serverfault.com/questions/52034/what-is-the-difference-between-double-and-single-square-brackets-in-bash

File: configure-repo.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
```
Comment (darcywong00) [inline configure-repo.sh]:
```suggestion
#!/bin/bash
```

This got rid of the `[[` error I mention below

File: configure-repo.sh (hunk 2 above)
Comment (mcdurdin) [reply configure-repo.sh]: Read much of that link and TIL. `[[` FTW.

Comment (mcdurdin) [reply configure-repo.sh]: > btw even with the error, the rest of the script ran

Oops, should have `set -e` at the start of the script!

**Comment by mcdurdin** [comment]
@darcywong00 after installation, did you try doing a commit with a dodgy commit message to see if it worked?

**Pull request merged**